Updated to the current path structure

KathrynBaker opened the pull request:
My slightly hacked testing sees this pull in a macro from a config (that has been set via a GUI), it probably needs checking by the expert.

To test:
Have something which has a readable macro when running, check that a macro set via the gui (so stored in the iocs.xml for the configuration) is used in that position.

KathrynBaker committed:
Updated to the current path structure

FreddieAkeroyd commented:
Hi, this commit would break loading components reference in a config, but looking at the code further it seems that it is relying on the blockserver adding /configurations and /components to the relevant bits. I can only assume that it used to do this!

KathrynBaker commented:
I’m not surprised I missed a knock-on effect. We do, however, need to solve this, as at the moment and macros set via the GUI are not actioned, as they can’t be seen by the IOCs.
I didn’t realise that this code interacted with the blockserver, I had thought this was part of the init of an IOC, so whilst the blockserver would trigger it when starting an IOC via procserv it wouldn’t interact with it directly.
Should the change instead be on line 625 for the call to LoadIOCs (which doesn’t work without that extra configuration)?

Regards,
Kathryn

FreddieAkeroyd committed:
Fix loading of configs and components

It appears the underlying code expected the /configurations
and /components prefix to be provided by the blockserver. Maybe
this used to be the case? Now modified to hansdle this.

FreddieAkeroyd commented:
The code only interacts with the blockserver in that it reads files created by the blockserver, such as last_config.txt   It seems files like this, and the xml, used to (or were assumed to) have /configurations etc specified as a prefix

FreddieAkeroyd committed:
Fix sim level propagation
